utils/ffmpegvideodecode/ffmpeg_video_dec.cpp

```diff
@@ -0,0 +1,866 @@
+    num_chroma_planes_ = GetChromaPlaneCount(video_surface_format_);
+    if (p_video_format->chroma_format == rocDecVideoChromaFormat_Monochrome) num_chroma_planes_ = 0;
+    chroma_vstride_ = static_cast<int>(std::ceil(surface_vstride_ * GetChromaHeightFactor(video_surface_format_)));
+    // Fill output_surface_info_
+    output_surface_info_.output_width = target_width_;
+    output_surface_info_.output_height = target_height_;
+    output_surface_info_.output_pitch  = surface_stride_;
+    output_surface_info_.output_vstride = (out_mem_type_ == OUT_SURFACE_MEM_DEV_INTERNAL) ? surface_vstride_ : target_height_;
+    output_surface_info_.bit_depth = bitdepth_minus_8_ + 8;
+    output_surface_info_.bytes_per_pixel = byte_per_pixel_;
+    output_surface_info_.surface_format = video_surface_format_;
+    output_surface_info_.num_chroma_planes = num_chroma_planes_;
+    if (out_mem_type_ == OUT_SURFACE_MEM_DEV_COPIED) {
+        output_surface_info_.output_surface_size_in_bytes = GetFrameSize();
+        output_surface_info_.mem_type = OUT_SURFACE_MEM_DEV_COPIED;
+    } else if (out_mem_type_ == OUT_SURFACE_MEM_HOST_COPIED) {
+        output_surface_info_.output_surface_size_in_bytes = GetFrameSize();
+        output_surface_info_.mem_type = OUT_SURFACE_MEM_HOST_COPIED;
+    }
+
+    // If the coded_width or coded_height hasn't changed but display resolution has changed, then need to update width and height for
+    // correct output with cropping. There is no need to reconfigure the decoder.
+    if (!is_decode_res_changed && is_display_rect_changed) {
+        return 1;
+    }
+
+    input_video_info_str_.str("");
+    input_video_info_str_.clear();
+    input_video_info_str_ << "Input Video Resolution Changed:" << std::endl
+        << "\tCoded size   : [" << p_video_format->coded_width << ", " << p_video_format->coded_height << "]" << std::endl
+        << "\tDisplay area : [" << p_video_format->display_area.left << ", " << p_video_format->display_area.top << ", "
+            << p_video_format->display_area.right << ", " << p_video_format->display_area.bottom << "]" << std::endl;
+    input_video_info_str_ << std::endl;
+    is_decoder_reconfigured_ = true;
+    return 1;
+}
+
+/**
+ * @brief 
+ * 
+ * @param pPicParams 
+ * @return int 1: success 0: fail
+ */
+int FFMpegVideoDecoder::HandlePictureDecode(RocdecPicParams *pPicParams) {
+    AVPacket *av_pkt = av_packets_[av_pkt_cnt_];
+    std::pair<uint8_t *, int> *packet_data = &av_packet_data_[av_pkt_cnt_];
+    if (last_packet_.payload_size > packet_data->second) {
+        void *new_pkt_data = av_realloc(av_pkt->data, (last_packet_.payload_size + MAX_AV_PACKET_DATA_SIZE));  // add more to avoid frequence reallocation
+        if (!new_pkt_data) {
+            std::cerr << "ERROR: couldn't allocate packet data" << std::endl;
+        }
+        packet_data->first   = static_cast<uint8_t *> (new_pkt_data);
```

Comment:
minor formatting: remove the space `<uint8_t *> (new_pkt_data);` -> `<uint8_t *>(new_pkt_data);`